### ISSUE TITLE
Auto generate installation instructions for packages

### ISSUE THREAD

**Issue by JoshLove-msft** [Azure/azure-sdk-tools#3136]

**Comment by kurtzeborn** [comment]
@JoshLove-msft, can you provide more context here?  Are you talking about when these appear in our various readmes?

I worry that the work to automate this would be more difficult than simply adding "prerelease" if you want "beta".  This also seems very language specific as other languages have different tools and versioning systems.  Please reach out to @weshaggard if we're missing something.

**Comment by JoshLove-msft** [comment]
I agree that the validation may differ per language, but I was thinking that something like this could live in the Verify-Readme script and have language-specific logic. I think all languages include installation instructions in one form or another in the Readme.

I've implemented something for .NET that validates there are GA instructions if the library has a GA version.

**Comment by weshaggard** [comment]
@JoshLove-msft what did you do for the .NET repo? Also for these install instructions wouldn't it be easier to have simple instructions with a footnote to add `--prerelease` if you want the latest beta?

**Comment by JoshLove-msft** [comment]
> @JoshLove-msft what did you do for the .NET repo?

See https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/CodeChecks.ps1#L107-L159

> Also for these install instructions wouldn't it be easier to have simple instructions with a footnote to add `--prerelease` if you want the latest beta?

We could do that, but it is somewhat annoying for the case where there is no GA release yet, hence the desire for a more customized installation experience.

**Comment by weshaggard** [comment]
Yes it is kind of annoying for a little while but I don't think that will hit very many people and it will make it such that these steps will be copy/paste and we don't have to worry about trying to update maintain them at all.

**Comment by JoshLove-msft** [comment]
Well we would still need some sort of checks to ensure people are pasting in the installation instructions correctly, so I'm inclined to stick with what we have for .NET as it provides a slightly better user experience.

**Comment by weshaggard** [comment]
What about cases where there are both GA and prereleases? Do you plan to have a note about prerelease flag? 

We have some basic readme validation which verifies that people have the correct headers but it might be interesting to think about a little deeper templating mechanism that would enable folks to add a set of common text like this for the sections as well but I'm not sure what that would look like currently. I think for now keeping what you have in .NET until a clearer pattern or scenario emerges will be the best option for this currently.

**Comment by JoshLove-msft** [comment]
If there are both GA and prereleases, we would only require that GA instructions are there. In an ideal world, I think this would be auto-generated into the Readme based on the changelog entries.

**Comment by weshaggard** [comment]
I actually think it is less about the changelog but more about the version we have defined for a given package. The trouble is we don't currently have anything that parses and can update READMEs as part of the process. If you would like you could attempt to add this as part of https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/Update-PkgVersion.ps1 which is what we use when we update or set the package versions. However that does mean you would need to do more logic to determine if we ever shipped a GA version or not, I suppose that might be what you were meaning by using the changelog which could be one way of doing this, we could also consider using the release tags to figure out the versions that have shipped as well.
